### PR TITLE
Patch users

### PR DESCRIPTION
**Pull request by aviehu**

### CODE REVIEW

File: lib/app/index.js [hunk 1]
```diff
@@ -5,16 +5,19 @@ import configGetUsers from '../handlers/configGetUsers';
 import configPostUsers from '../handlers/comfigPostUsers';
 import getConnection from '../models/createConnection';
 import getUsers from '../models/createUsers';
+import configUpdateUsers from '../handlers/configUpdateUsers';
```
Comment (roeeyud) [inline lib/app/index.js]: configurePatchUsers...
Don't change terminology back and forth

File: lib/app/index.js [hunk 2]
```diff
@@ -5,16 +5,19 @@ import configGetUsers from '../handlers/configGetUsers';
 import configPostUsers from '../handlers/comfigPostUsers';
```
Comment (roeeyud) [inline lib/app/index.js]: Missing new test script!!!!!

File: lib/handlers/configPatchUsers.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+describe('patchUsers test', () => {
+  beforeEach(async () => {
+    await users.sync();
+  });
+
+  afterEach(() => users.destroy({
+    truncate: true,
+  }));
+
+  const patchUsers = configPatchUsers(users);
```
Comment (roeeyud) [inline lib/handlers/configPatchUsers.test.js]: move to top under `const users = ...`

File: lib/handlers/configPatchUsers.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+function configPatchUsers(users) {
+  return async function patchUsers(req, res) {
+    const values = { userName: req.body.userName, age: req.body.age };
+    const condition = { where: { id: req.body.id } };
+    const options = { multi: true };
+    if (typeof (values.userName) === 'string' && typeof (values.age) === 'number') {
```
Comment (roeeyud) [inline lib/handlers/configPatchUsers.js]: Does Sequalize do this check for you?

File: lib/handlers/configPatchUsers.test.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    await postUsers(data);
+
+    await patchUsers(newData, res);
+
+    const usersModels = await users.findAll({
+      attributes: ['userName', 'age'],
+    });
+    const result = await usersModels.map(model => model.get());
+
+    expect(result).toEqual([{ userName: 'Roee', age: 30 }]);
```
Comment (roeeyud) [inline lib/handlers/configPatchUsers.test.js]: Where is `expect(res.send).....`

File: lib/handlers/configPatchUsers.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+function configPatchUsers(users) {
```
Comment (roeeyud) [inline lib/handlers/configPatchUsers.js]: Change file name to `configPatchUser`
you can only patch a single user with this function

File: lib/handlers/configPatchUsers.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  return async function patchUsers(req, res) {
+    const values = { userName: req.body.userName, age: req.body.age };
+    const condition = { where: { id: req.body.id } };
+    const options = { multi: true };
+    if (typeof (values.userName) === 'string' && typeof (values.age) === 'number') {
+      await users.sync();
+      await users.update(values, condition, options);
+      return res.sendStatus(200);
```
Comment (roeeyud) [inline lib/handlers/configPatchUsers.js]: you should send back the new user

File: lib/handlers/configPatchUsers.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    }
+    return res.sendStatus(400);
```
Comment (roeeyud) [inline lib/handlers/configPatchUsers.js]: 400 is OK but I want to send also a message to display for the user

File: package.json [hunk 9]
```diff
@@ -25,6 +25,7 @@
     "body-parser": "^1.18.3",
     "eslint-config-airbnb": "^17.1.0",
     "express": "^4.16.3",
+    "express-subdomain": "^1.0.5",
```
Comment (roeeyud) [inline package.json]: remove

File: lib/handlers/configPatchUsers.js (hunk 4 above)
Comment (aviehu) [reply lib/handlers/configPatchUsers.js]: right now it doesnt i can add validation

File: lib/handlers/configPatchUsers.js (hunk 8 above)
Comment (aviehu) [reply lib/handlers/configPatchUsers.js]: like an alert or something like that?

File: lib/handlers/configPatchUsers.test.js (hunk 5 above)
Comment (aviehu) [reply lib/handlers/configPatchUsers.test.js]: expect(res.sendStatus).toHaveBeenCalledWith(400); i have it like this

File: lib/handlers/configPatchUsers.js (hunk 7 above)
Comment (aviehu) [reply lib/handlers/configPatchUsers.js]: instead of the status?

File: lib/handlers/configPatchUser.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+    const values = { userName: req.body.userName, age: req.body.age };
+    const condition = { where: { id: req.body.id } };
+    const options = { multi: true };
+    try {
+      await users.sync();
+      await users.update(values, condition, options);
+    } catch (error) {
+      return res.send(error);
+    }
+    return res.send(req.body);
```
Comment (roeeyud) [inline lib/handlers/configPatchUser.js]: not good... this I'd expect to see this response from sequalize

File: lib/handlers/configPostUsers.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
+
+function configPostUsers(users) {
+  return async function postUsers(req, res) {
+    try {
+      await users.sync();
+      await users.create(req.body);
+    } catch (error) {
+      return res.send(error);
+    }
+    return res.send(req.body);
```
Comment (roeeyud) [inline lib/handlers/configPostUsers.js]: this response will not send back the new ID

Comment (aviehu) [reply lib/handlers/configPostUsers.js]: i guess i can count the number of instances and just send that back as id

**Pull request merged**